### PR TITLE
Feature/floating islands

### PR DESCRIPTION
**Pull request by Jonpro03**
* Added a floating island generator
* Moved worldgen decorator features out of the Overworld namespace
* Fixed issue where new players would not spawn at world spawn
* Fixed issue with client timeout waiting for 14x14 chunks at join
* Fixed issue not sending clients all chunks in their view distance
* Fixed issue with biome palette not being read/written correctly

### CODE REVIEW

File: Obsidian/WorldData/World.cs [hunk 1]
```diff
@@ -819,6 +819,7 @@ internal async Task SetWorldSpawnAsync()
                 }
             }
         }
+        Server.Logger.LogInformation($"Failed to set World Spawn.");
```
Comment (inftord) [inline Obsidian/WorldData/World.cs]: should be LogError or LogWarn IMO

Comment (Jonpro03) [reply Obsidian/WorldData/World.cs]: Concur

File: Obsidian/Client.cs [hunk 2]
```diff
@@ -530,7 +530,7 @@ await QueuePacketAsync(new LoginSuccess(Player.Uuid, Player.Username)
 
         await SendPlayerListDecoration();
         await SendPlayerInfoAsync();
-        await Player.UpdateChunksAsync(distance: 7);
+        while (!await Player.UpdateChunksAsync(distance: 7)) { Thread.Sleep(1); }
```
Comment (Tides) [inline Obsidian/Client.cs]: Why Thread.Sleep and not Task.Delay 🤨

Comment (Craftplacer) [inline Obsidian/Client.cs]: This line seems sus.

File: Obsidian/Entities/Player.cs [hunk 3]
```diff
@@ -942,8 +942,9 @@ private async Task PickupNearbyItemsAsync(float distance = 0.5f)
     /// <param name="unloadAll"></param>
     /// <param name="distance"></param>
     /// <returns></returns>
```
Comment (Craftplacer) [inline Obsidian/Entities/Player.cs]: Update XMLdoc accordingly

**Review by Craftplacer — DISMISSED** [review]

File: Obsidian/Client.cs (hunk 2 above)
Comment (Jonpro03) [reply Obsidian/Client.cs]: Yeah, I agree... this is a bad idea

**Review by Seb-stian — APPROVED** [review]

**Review by Tides — APPROVED** [review]

File: Obsidian/Client.cs (hunk 2 above)
Comment (Jonpro03) [reply Obsidian/Client.cs]: It was late. I was drunked the beers

**Pull request merged**